### PR TITLE
Gtfs loader v2 jsonification

### LINKED ISSUE #1533: GTFS Schedule Pipeline: gtfs_loader core updates
Create a new gtfs_loader DAG that can (AC):

- [x] Read in zipped GTFS data (from #1528)
- [x] Unzip contents and save as JSONL in hive partitioned paths based on:
  - [x] The date and time that the code actually runs, not the Airtable execution date
  - [x] UUID (not ITP ID & URL number)
- [x] Outputs a single outcomes file summarizing download outcomes like what was implemented in https://github.com/cal-itp/data-infra/pull/1458 for RT
- [x] This DAG should run every day on all downloaded data (no pre-assessment of whether data changed)
Will need handling for CSV parse errors.

### PR DESCRIPTION
**Pull request by lauriemerrell**
# Description

Completes #1533, building on the work in #1696. Specifically, reads in files from unzipped GTFS downloads, converts them `jsonl`, and saves the `jsonl` files in the same hive partitions as the input file & creates accompanying outcomes file.

This PR does *not* create external tables for the JSONL data, that will be handled in #1536.

Resolves #1533

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

Run in local Airflow for 8/12 (date that has test data available.)

## Screenshots 

_Top two (existing) tasks manually marked as failures just to speed things up_
image

### CODE REVIEW

File: airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/METADATA.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+tooltip: "Convert known GTFS files to JSONL"
```
Comment (atvaccaro) [inline airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/METADATA.yml:1]: neat!

File: airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/agency.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+operator: operators.GtfsGcsToJsonlOperator
+
+date: "{{ execution_date }}"
```
Comment (atvaccaro) [inline airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/agency.yml:3]: is this needed? can operators get the execution date in the task context?

File: airflow/plugins/operators/gtfs_csv_to_jsonl.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    result = ScheduleParseResult(filename="results.jsonl", dt=day, outcomes=outcomes)
+    result.save(fs)
+
+    assert len(files) == len(
+        result.outcomes
+    ), f"ended up with {len(outcomes)} outcomes from {len(files)} files"
+
+
+class GtfsGcsToJsonlOperator(BaseOperator):
+    def __init__(self, date, input_table, gtfs_filename, *args, **kwargs):
```
Comment (atvaccaro) [inline airflow/plugins/operators/gtfs_csv_to_jsonl.py:146]: ah looks like `date` isn't actually used?

File: airflow/plugins/operators/gtfs_csv_to_jsonl.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        self.input_table = input_table
+        self.gtfs_filename = gtfs_filename
```
Comment (atvaccaro) [inline airflow/plugins/operators/gtfs_csv_to_jsonl.py:148]: is there a situation where these would ever differ except for removing the file extension?

**Review by atvaccaro — CHANGES_REQUESTED** [review]
do we want to set a job failure threshold anywhere?

**Comment by lauriemerrell** [comment]
> do we want to set a job failure threshold anywhere?

What do you mean? Like if >X% of outcomes are failures, we fail the job? Open to it but not sure what we'd set the threshold at

File: airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/agency.yml (hunk 2 above)
Comment (lauriemerrell) [reply airflow/dags/unzip_and_validate_gtfs_schedule/convert_to_json/agency.yml]: ope yeah good catch, this was a vestige from a prior attempt. fixed https://github.com/cal-itp/data-infra/pull/1738/commits/7b05d2e3fa8141381c9ee04ea1dd5265d84dcb18

File: airflow/plugins/operators/gtfs_csv_to_jsonl.py (hunk 3 above)
Comment (lauriemerrell) [reply airflow/plugins/operators/gtfs_csv_to_jsonl.py]: yeah see above, fixed https://github.com/cal-itp/data-infra/pull/1738/commits/7b05d2e3fa8141381c9ee04ea1dd5265d84dcb18

File: airflow/plugins/operators/gtfs_csv_to_jsonl.py (hunk 4 above)
Comment (lauriemerrell) [reply airflow/plugins/operators/gtfs_csv_to_jsonl.py:148]: yeah, the case where something has a wrong name in the input but we want to map it onto the valid place. so like `_stops.txt` in input

Comment (lauriemerrell) [reply airflow/plugins/operators/gtfs_csv_to_jsonl.py:148]: I don't know that this has actually happened, but it's something that this architecture specifically allows for

Comment (atvaccaro) [reply airflow/plugins/operators/gtfs_csv_to_jsonl.py:148]: Ok cool, I think providing a default could allow for less boilerplate but seems fine

**Comment by atvaccaro** [comment]
> 

Yeah really it's to catch situations where a bug broadly impacted the job and/or we think that the overall success rate is oddly low and worth "failing" the job. It can occur after the results upload so it's primarily something to bubble up through alerting. The downloader is 95%, maybe this could be similar?

**Review by atvaccaro — APPROVED** [review]

File: airflow/plugins/operators/gtfs_csv_to_jsonl.py (hunk 4 above)
Comment (lauriemerrell) [reply airflow/plugins/operators/gtfs_csv_to_jsonl.py]: yeah fair, https://github.com/cal-itp/data-infra/pull/1738/commits/66eda24220ed159db29a4b9892b86aadc5b21984

**Pull request merged**